### ISSUE TITLE
Can't install in a subpath anymore

### ISSUE THREAD

**Issue by Salamandar** [OliveTin/OliveTin#404]
Before https://github.com/OliveTin/OliveTin/pull/384 (i think?) I was able to host olivetin at `mydomain.tld/olivetin/` with the proper nginx and olivetin configuration.

Now it's not possible anymore, due to absolute paths to `/index.<shasum>.js`, logos, and websocket urls.

Could you please make possible again to host olivetin in a subdir ? Thanks <3

**Comment by jamesread** [comment]
Hey @Salamandar , yeah, unfortunately this changed back in August. This is the reason: https://docs.olivetin.app/upgrade-notes.html#_2024_08_14

How does this affect you, can you work around it?

**Comment by Salamandar** [comment]
It does not affect me especially but I'm packaging Olivetin for Yunohost (see https://github.com/yunohost-apps/olivetin_ynh) and we were allowing subpath installs till now 😭

Do you think it's possible to allow that again ? It would be really nice. It's a bit annoying to have to handle a subdomain+nginx config+certificates just for olivetin 😅

**Comment by pomazanbohdan** [comment]
I am deploying odoo on customer subdomains, but don't have the ability to use other subdomains, I use for example /pgadmin4 for database management. 
I wanted to use OliveTin on /control to allow me to overload the container or make pulling improvements from the outside easy for developers.

**Comment by stellarpower** [comment]
I am installing alongside a larger application, and using this to perform authentication before proxy-passing to OliveTin. I generally use subdomains because of all the potential problems, so understand the complexity issues. But just for context, in this case I want to be sure that the cookie is sent along to the larger application for checking my user is logged in, so using a path is probably easier than a sub-subdomain or trying to juggle things somehow.

**Comment by jamesread** [comment]
Heya, so be clear, subdomains are absolutely supported: olivetin.example.com/ or ot.customer.foo.example.com/

However, I dropped support for sub paths, like example.com/apps/olivetin and customer1.foo.bar.com/ot/

**Comment by jamesread** [comment]
I will try and bring back sub path support, now that the new navigation system is stable. 

Obviously I need to set that in cookies too. 

It would be good to know how much demand there is for this though, so please leave comments about why sub paths are useful for your use case.

**Comment by stellarpower** [comment]
Yes, got you - only that subdomains don't work so well in the larger infrastructure where I'm deploying OliveTin in this specific instance; otherwise I always use them when deploying an application because I have wildcard certs already and the proxy configs are just such a pain in the arse at the best of times. So in this case OliveTin doesn't have subpath whilst outside infrastructure doesn't have subdomain. I just pulled an older image though so wasn't an issue in and of itself. Just figured you might like the context, as you say, for my "user story". A subpath is a cheap way of making sure I receive a cookie used for another application and then I can have my proxy check the authentication before passing to OliveTin to make sure that only logged-in users are able to access the dashboard. Whilst I wouldn't put anything critical on there, it's a quick hack to stop it from being accessible to the world at large.

**Comment by TomHellier** [comment]
Subpaths are useful for my use case (I'm just looking at olivetin), I think I want to run it on some kubernetes clusters so I can use it to expose some kubernetes administrative actions to various users, and the dns is easier if we just have it on (mycluster.example.com/olivetin/)

**Comment by Pool0513** [comment]
Please, yes add subpaths :)

**Comment by jamesread** [comment]
Thanks for your patience on this one, everyone.

Subpath support won’t be restored in OliveTin 2k. I do want to queue this up to look at again in the OliveTin 3k version.

If this is still important for your setup, feel free to add any extra detail about your use case or current workaround.

_Disclaimer: I’m Zeratul, an AI bot assisting with issue maintenance. A human maintainer can review or correct this guidance._

**Comment by jamesread** [comment]
I've taken a long hard look at this one in 3k, and it's just a huge amount of effort, both to change and to maintain. 

It's not just the frontend webui, it means changing all the following paths and making sure they work under different URLs;

- /oauth/login
- /oauth/callback
- /readyz
- /webhooks
- /theme.css (the theme loader system)
- /custom-webui/
- / - the main frontend

Then there's getting the frontend to use all those paths reliably, making sure all the cookies work, etc it's just an ongoing headache.

I try really hard to make everybody happy, and I've read through the comments, but this is just one of those times that for good technical reasons I cannot implement this and take on the cost of maintaining it again in 3k. It looks like this has requests from 5 people over the last 2 years - that's not enough to justify the technical work.

Thanks for raising the issue, I know this will annoy some people, but the cost outweighs the benefits for too few people here.
